site/react/ssr.md

```diff
@@ -0,0 +1,83 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: Server Side Rendering
+  - - meta
+    - name: description
+      content: An overview on how to use React Hooks with Server Side Rendering
+  - - meta
+    - property: og:description
+      content: An overview on how to use React Hooks with Server Side Rendering
+  - - meta
+    - name: twitter:title
+      content: Server Side Rendering
+  - - meta
+    - name: twitter:description
+      content: An overview on how to use React Hooks with Server Side Rendering
+---
+
+# Server Side Rendering (SSR)
+
+When using the React hooks exported by Account Kit in a server-side rendered setting, you will see inconsistencies of the user state between the server and the client. This will lead to flashes of content when a user is logged in. To avoid this, the account state can be optimistically loaded on the server and passed to the client.
+
+To enable this setting, you can set `ssr: true` when creating a config.
+
+```ts
+import { createConfig } from "@alchemy/aa-alchemy/config";
+import { sepolia } from "@alchemy/aa-core";
+
+export const config = createConfig({
+  // required
+  rpcUrl: "/api/rpc",
+  chain: sepolia,
+  ssr: true, // [!code ++]
+});
+```
+
+This setting will defer hydration of the account state to the client after the intial mount.
+
+## Persisting the Account State
+
+### Cookie Storage
+
+To consistently pass the state between the server and the client, you can pass in a cookie storage to the `config` object created above.
```

Comment:
I'm pretty unfamiliar with SSR and have never actually used it myself, so maybe this would be obvious to someone who knows how SSR works. But from reading this I don't really know what using cookie storage is actually doing. Does this mean that the server is setting cookies on the client to maintain some kind of session, or is the client setting the cookies as a way to send the server data that the server can use to render?

Reply:
no these are great questions regardless of familiarity with ssr. I've elaborated more in the doc now to explain what exactly the storage is used for. But essentially the cookie storage allows the client to write a cookie which it can then pass to the server. The state manager can then use this initial state on the server for certain pieces of state like the user's logged in status to ensure that what is rendered on the server matches the client